### DISCUSSION TITLE
Version 0.0.17

### DISCUSSION THREAD

**Discussion by Ryochan7** [Ryochan7/DS4MapperTest#31] [Announcements]
Fixes, Steam Controller support

.NET 8 Desktop Runtime x64
https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/runtime-desktop-8.0.1-windows-x64-installer

* Updated FakerInputDll files
* Set DataContext to null early for some windows
* Added initial Steam Controller support
* Make Steam Controller support use some custom action defaults when editing a profile
* Included some example Steam Controller profiles
* Fixed trackball friction not taking affect when changed
* Do not force resetting of output stick coordinates if Gyro Mouse Joystick is not active
* Remove traces of AnyCPU target from project
* Added Daggerfall example profile to project
* Added extra attempts for reading a serial early for a BLE Steam Controller connection
* Added test battery level reading for BLE Steam Controller connection
* Change SC BT reader previous timestamp before potential PING packet processing
* Use newer action processing for mapper interaction
* Added unit test project to solution
* Various edits to test KB+M focused profiles. Make new project configurations to force skipping outputting events
* Make sure single layer is being examined when switching mapped Gyro action
* Keep explicit reference to most recently applied base Action Layer
* Removed use of now redundant ManualResetEventSlim instances in VMs

<hr /><em>This discussion was created from the release <a href='https://github.com/Ryochan7/DS4MapperTest/releases/tag/v0.0.17'>Version 0.0.17</a>.</em>

**Comment by ILXVITATE** [comment]
i was reading a now archived post about adding support for the scuff envision pro, i was under the impression it couldnt be done because scuf only wants you to use the icue software. im here to ask can it actually be done? Do i need to gift you a scuf envision pro to figure it out? im willing.

**Reply by Ryochan7** [reply]
I do not remember anything about the SCUF envision controller. Not sure if anyone has figured out how the packet payloads work. Either way, I do not think I would be able to do anything about it. I was wanting to tweak this software a bit more but problems keep getting in the way. I have hardly even played any video games in the last few months.
